### ISSUE TITLE
Exception Handling Flaw in streaming.py

### ISSUE THREAD

**Issue by muse-enthusiast** [tweepy/tweepy#734]
The tweepy.streaming.py file is missing proper exception handing in line 313. See Traceback: 

```
Traceback (most recent call last):
   ...
  File "/usr/local/lib/python3.4/dist-packages/tweepy/streaming.py", line 445, in filter
    self._start(async)
  File "/usr/local/lib/python3.4/dist-packages/tweepy/streaming.py", line 361, in _start
    self._run()
  File "/usr/local/lib/python3.4/dist-packages/tweepy/streaming.py", line 294, in _run
    raise exception
  File "/usr/local/lib/python3.4/dist-packages/tweepy/streaming.py", line 263, in _run
    self._read_loop(resp)
  File "/usr/local/lib/python3.4/dist-packages/tweepy/streaming.py", line 313, in _read_loop
    line = buf.read_line().strip()
AttributeError: 'NoneType' object has no attribute 'strip'

```

**Comment by mguida22** [comment]
Checkout #576 it seems like the same issue

**Comment by drevicko** [comment]
Looking at the code, it seems ReadBuffer.read_line() is returning None, which boils down to it's _pop() method returning None. This should be the result of `r.decode(self._encoding)` where `r` ought to be a bytes object

**Comment by drevicko** [comment]
This should be fixed by f4bfce5
